src/commands/tileindex-validate/tileindex.validate.ts

```diff
@@ -357,6 +362,21 @@ export interface TiffLocation {
   bands: string[];
 }
 
+/**
+ * Reproject the bounding box if the source and target projections are different.
+ * @param bbox input bounding box
+ * @param sourceProjection CRS of the input bounding box
+ * @param targetProjection target CRS
+ */
+export function reprojectIfNeeded(bbox: BBox, sourceProjection: Projection, targetProjection: Projection): BBox | null {
+  {
+    if (targetProjection === sourceProjection) return bbox;
+    const [ulX, ulY] = targetProjection.fromWgs84(sourceProjection.toWgs84([bbox[0], bbox[3]])) as [number, number]; // Typescript inferred this as number[] but should be [number, number] | [number, number, number]
```

Comment:
This comment is a little off, the typing of the function specifies number[] it could be refined to [number, number] when [number, number] is passed in, so the casting is fine

Reply:
I guess that's what the comment was trying to say, I have refined it from 
`// Typescript inferred this as number[] but should be [number, number] | [number, number, number]`
to
`// Function is typed as number[] but could be refined to [number, number] | [number, number, number]. With 2 input args, it will always return [number, number]`